wear/src/main/java/io/homeassistant/companion/android/tiles/ThermostatTile.kt

```diff
@@ -0,0 +1,325 @@
+        val tileConfig = thermostatTileDao.get(tileId)
+
+        if (requestParams.currentState.lastClickableId.isNotEmpty()) {
+            if (wearPrefsRepository.getWearHapticFeedback()) hapticClick(applicationContext)
+        }
+
+        val freshness = when {
+            (tileConfig?.refreshInterval != null && tileConfig.refreshInterval!! <= 1) -> 0
+            tileConfig?.refreshInterval != null -> tileConfig.refreshInterval!!
+            else -> DEFAULT_REFRESH_INTERVAL
+        }
+
+        val tile = Tile.Builder()
+            .setResourcesVersion("$TAG$tileId.${System.currentTimeMillis()}")
+            .setFreshnessIntervalMillis(TimeUnit.SECONDS.toMillis(freshness))
+
+        if (!serverManager.isRegistered()) {
+            tile.setTileTimeline(
+                loggedOutTimeline(
+                    this@ThermostatTile,
+                    requestParams,
+                    R.string.thermostat,
+                    R.string.thermostat_tile_log_in
+                )
+            ).build()
+        } else {
+            if (tileConfig?.entityId.isNullOrBlank()) {
+                tile.setTileTimeline(
+                    Timeline.fromLayoutElement(
+                        LayoutElementBuilders.Box.Builder()
+                            .addContent(
+                                LayoutElementBuilders.Text.Builder()
+                                    .setText(getString(R.string.thermostat_tile_no_entity_yet))
+                                    .setMaxLines(10)
+                                    .build()
+                            ).build()
+                    )
+                ).build()
+            } else {
+                try {
+                    val entity = tileConfig.entityId?.let {
+                        serverManager.integrationRepository().getEntity(it)
+                    }
+                    check(entity != null)
+
+                    val lastId = requestParams.currentState.lastClickableId
+                    var targetTemp = tileConfig.targetTemperature ?: entity.attributes["temperature"]?.toString()?.toFloat()
+
+                    val config = serverManager.webSocketRepository().getConfig()
+                    val temperatureUnit = config?.unitSystem?.getValue("temperature").toString()
+
+                    if (targetTemp != null && (lastId == TAP_ACTION_UP || lastId == TAP_ACTION_DOWN)) {
+                        val entityStr = entity.entityId
+                        var stepSize = entity.attributes["target_temp_step"]
+
+                        stepSize = if (stepSize == null && temperatureUnit == "°F") {
+                            1.0f
+                        } else if (stepSize == null) {
+                            0.5f
+                        } else {
+                            stepSize.toString().toFloat()
+                        }
+
+                        val updatedTargetTemp = targetTemp + if (lastId == TAP_ACTION_UP) +stepSize.toFloat() else -stepSize.toFloat()
```

Comment:
Try to avoid a `var` if you immediately reassign it, and don't convert floats to floats again ;)

```suggestion
                        val attrStepSize = (entity.attributes["target_temp_step"] as? Number)?.toFloat()
                        val stepSize = attrStepSize ?: if (temperatureUnit == "°F") 1.0f else 0.5f
                        val updatedTargetTemp = targetTemp + if (lastId == TAP_ACTION_UP) +stepSize else -stepSize
```

Reply:
Well, that's definitely a lot cleaner!

Reply:
[7803458](https://github.com/home-assistant/android/pull/4959/commits/780345872ec83a5ec206012289215ff631b69f41)